diff --git a/Modules/grpmodule.c b/Modules/grpmodule.c
--- a/Modules/grpmodule.c
+++ b/Modules/grpmodule.c
@@ -264,6 +264,10 @@ grp_getgrall_impl(PyObject *module)
 {
     PyObject *d;
     struct group *p;
 
+#ifdef __ANDROID__
+    return PyList_New(0);
+#else
     if ((d = PyList_New(0)) == NULL)
         return NULL;
     setgrent();
@@ -281,6 +285,7 @@ grp_getgrall_impl(PyObject *module)
     }
     endgrent();
     return d;
+#endif
 }
 
 static PyMethodDef grp_methods[] = {
